metrics/prometheus.go

```diff
@@ -2,14 +2,31 @@ package metrics
 
 import (
 	"flag"
+	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
+	dto "github.com/prometheus/client_model/go"
 	"log"
 	"net/http"
 	"strconv"
 )
 
 type PrometheusConfig struct {
+	//keep track of last source and sink details for lag calculation
+	lastSourceDetail map[string]map[int32]int64
+	lastSinkDetail map[string]map[int32]int64
+
+	//metric collectors
+	sourceOffMetric *prometheus.GaugeVec
+	sinkOffMetric *prometheus.GaugeVec
+	lagOffMetric *prometheus.GaugeVec
+	partitionOwned *prometheus.GaugeVec
```

Comment:
we should move these outside config object, I believe we don't expected users to define this